include/SFML/Window/Vulkan.hpp

```diff
@@ -0,0 +1,114 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2019 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_VULKAN_HPP
+#define SFML_VULKAN_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Export.hpp>
+#include <SFML/Window/WindowHandle.hpp>
+#include <vector>
+#include <stdint.h>
+
+
+typedef struct VkInstance_T* VkInstance;
+
+#if defined(__LP64__) || defined(_WIN64) || (defined(__x86_64__) && !defined(__ILP32__) ) || defined(_M_X64) || defined(__ia64) || defined (_M_IA64) || defined(__aarch64__) || defined(__powerpc64__)
+
+typedef struct VkSurfaceKHR_T* VkSurfaceKHR;
+
+#else
+
+typedef uint64_t VkSurfaceKHR;
+
+#endif
+
+struct VkAllocationCallbacks;
+
+
+namespace sf
+{
+
+typedef void (*VulkanFunctionPointer)();
+
+////////////////////////////////////////////////////////////
+/// \brief Vulkan helper functions
+///
+////////////////////////////////////////////////////////////
+class SFML_WINDOW_API Vulkan
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Tell whether or not the system supports Vulkan
+    ///
+    /// This function should always be called before using
+    /// the Vulkan features. If it returns false, then
+    /// any attempt to use Vulkan will fail.
+    ///
+    /// If only compute is required, set \a requireGraphics
+    /// to false to skip checking for the extensions necessary
+    /// for graphics rendering.
+    ///
+    /// \param requireGraphics
+    ///
+    /// \return True if Vulkan is supported, false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    static bool isAvailable(bool requireGraphics = true);
```

Comment:
I find `requireGraphics` a bit ambiguous in some regards. If I set it to false, will extensions still be loaded if available, because I don't *require* them, so they may still be loaded.

Maybe `requestGraphics` or `includeGraphics` or a more explicit documentation comment would work better here.

Reply:
Our goal shouldn't be to teach people how Vulkan works. We should be able to assume that anybody who dares to touch Vulkan has some kind of idea of what they are doing.

This function doesn't actually load any extensions, because simply put, you don't _load_ extensions in Vulkan, you _require them to be enabled_ in the instance that you create. Instance creation is all or nothing, this is why you have to check beforehand if instance creation could even succeed with the required extensions enabled before going ahead and creating it.

For people who know how to work with Vulkan, the current naming/documentation should be enough for them to understand the usage.